consult-flymake.el

```diff
@@ -74,18 +69,36 @@
                            text)
                    'consult--candidate marker
                    'consult--type narrow))
-     (sort diags
-           (pcase-lambda (`(_ _ ,t1 _ ,m1 _) `(_ _ ,t2 _ ,m2 _))
-             (let ((s1 (flymake--severity t1))
-                   (s2 (flymake--severity t2)))
-               (or (> s1 s2) (and (= s1 s2) (< m1 m2)))))))))
+     ;; Sort first by position and then by the name of the buffer.
+     (sort
+      (sort diags
+            (pcase-lambda (`(,b1 _ ,t1 _ ,m1 _) `(,b2 _ ,t2 _ ,m2 _))
+              (let ((s1 (flymake--severity t1))
+                    (s2 (flymake--severity t2)))
+                (or (> s1 s2) (and (= s1 s2) (< m1 m2))))))
+      (pcase-lambda (`(,b1 _ _ _ _ _) `(,b2 _ _ _ _ _))
+        (string-lessp b1 b2))))))
```

Comment:
I am not fond of sorting twice. Please change the predicate instead.

Reply:
See my comment https://github.com/minad/consult/pull/589#discussion_r889033252. It is an exercise for you. I insist on this :)

Reply:
Okay, got it in one try :tada:, ISTG I actually tried this out but then some entries with a given mark were sorted before others even though order wise they shouldn't have been. I'm guessing I just misread it (I was testing with buffer names that were too similair to each other so they were easy to mistake).